rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -734,17 +755,68 @@ impl Controller {
                     )?;
                 }
                 if let Err(error) = self.refresh_system_tray_menu() {
-                    tracing::error!(?error, "Failed to refresh Resource list");
+                    tracing::error!(?error, "Failed to refresh menu");
                 }
 
-                self.update_disabled_resources().await?;
+                Ok(())
+            }
+        }
+    }
 
+    async fn handle_connect_result(
+        &mut self,
+        result: Result<(), IpcServiceError>,
+    ) -> Result<(), Error> {
+        let (first_time, start_instant, token) = match &self.status {
+            Status::Disconnected
+            | Status::TunnelReady { .. }
+            | Status::WaitingForNetwork { .. }
+            | Status::WaitingForTunnel { .. } => {
+                tracing::error!("Impossible logic error, received `ConnectResult` when we weren't waiting on the Portal connection.");
+                return Ok(());
+            }
+            Status::WaitingForPortal {
+                first_time,
+                start_instant,
+                token,
+            } => (
+                *first_time,
+                *start_instant,
+                token.expose_secret().clone().into(),
+            ),
+        };
+
+        match result {
+            Ok(()) => {
+                ran_before::set().await?;
+                self.status = Status::WaitingForTunnel { start_instant };
+                if let Err(error) = self.refresh_system_tray_menu() {
+                    tracing::error!(?error, "Failed to refresh menu");
+                }
                 Ok(())
             }
-            IpcServerMsg::TerminatingGracefully => {
-                tracing::info!("Caught TerminatingGracefully");
-                self.tray.set_icon(system_tray::Icon::SignedOut).ok();
-                Err(Error::IpcServiceTerminating)
+            Err(IpcServiceError::PortalConnection(error)) => {
+                tracing::warn!(
+                    ?error,
+                    "Failed to connect to Firezone Portal, will try again when the network changes"
+                );
+                if first_time {
+                    os::show_notification(
+                        "Firezone offline",
```

Comment:
```suggestion
                        "Internet offline",
```

I wonder if people will think this means the Firezone service is offline